prettyprinter-configurable/src/Text/PrettyBy/Internal.hs

```diff
@@ -574,256 +617,339 @@ application is implemented.
 
 -- | Throw @err@ when @b@ is stuck.
 type family ThrowOnStuck err (b :: Bool) :: Bool where
-    ThrowOnStuck _   'True  = 'True
-    ThrowOnStuck _   'False = 'False
-    ThrowOnStuck err _      = err
+  ThrowOnStuck _ 'True = 'True
+  ThrowOnStuck _ 'False = 'False
+  ThrowOnStuck err _ = err
+
+-- We have to use a type family here rather than a type alias,
+-- because otherwise it evaluates too early.
 
--- We have to use a type family here rather than a type alias, because otherwise it evaluates too early.
 -- | The error thrown when @HasPrettyDefaults config@ is stuck.
 type family HasPrettyDefaultsStuckError config :: Bool where
-    HasPrettyDefaultsStuckError config = TypeError
-        (     'Text "No ’HasPrettyDefaults’ is specified for " ':<>: 'ShowType config
-        ':$$: 'Text "Either you're trying to derive an instance, in which case you have to use"
-        ':$$: 'Text "  standalone deriving and need to explicitly put a ‘PrettyDefaultBy config’"
-        ':$$: 'Text "  constraint in the instance context for each type in your data type"
-        ':$$: 'Text "  that supports default pretty-printing"
-        ':$$: 'Text "Or you're trying to pretty-print a value of a type supporting default"
-        ':$$: 'Text "  pretty-printing using a config, for which ‘HasPrettyDefaults’ is not specified."
-        ':$$: 'Text "  If the config is a bound type variable, then you need to add"
-        ':$$: 'Text "    ‘HasPrettyDefaults <config_variable_name> ~ 'True’"
-        ':$$: 'Text "  to the context."
-        ':$$: 'Text "  If the config is a data type, then you need to add"
-        ':$$: 'Text "    ‘type instance HasPrettyDefaults <config_name> = 'True’"
-        ':$$: 'Text "  at the top level."
-        )
-
--- | A version of 'HasPrettyDefaults' that is never stuck: it either immediately evaluates
--- to a 'Bool' or fails with a 'TypeError'.
+  HasPrettyDefaultsStuckError config =
+    TypeError
+      ( 'Text "No ’HasPrettyDefaults’ is specified for " ':<>: 'ShowType config
+          ':$$: 'Text "Either you're trying to derive an instance, in which case you have to use"
+          ':$$: 'Text "  standalone deriving and need to explicitly put a ‘PrettyDefaultBy config’"
+          ':$$: 'Text "  constraint in the instance context for each type in your data type"
+          ':$$: 'Text "  that supports default pretty-printing"
+          ':$$: 'Text "Or you're trying to pretty-print a value of a type supporting default"
+          ':$$: 'Text
+                  "  pretty-printing using a config, \
+                  \for which ‘HasPrettyDefaults’ is not specified."
```

Comment:
I hate `fourmolu`. This way the error message code was formatted directly reflected how the error message would end up being shown to the user. Now it's just random fucking garbage.

Reply:
This is not fourmolu, I wrapped this long line (>100) manually.